rest/blip_sync.go

```diff
@@ -53,42 +53,43 @@ var kHandlersByProfile = map[string]func(*blipHandler, *blip.Message) error{
 
 // HTTP handler for incoming BLIP sync WebSocket request (/db/_blipsync)
 func (h *handler) handleBLIPSync() error {
-	if !h.server.GetDatabaseConfig(h.db.Name).Unsupported.Replicator2 {
-		return base.HTTPErrorf(http.StatusNotFound, "feature not enabled")
+	if c := h.server.GetConfig().ReplicatorCompression; c != nil {
+		blip.CompressionLevel = *c
 	}
 
+	// Create a BLIP context:
+	blipContext := blip.NewContext()
+	blipContext.Logger = func(fmt string, params ...interface{}) {
+		base.LogTo("BLIP", fmt, params...)
+	}
+	blipContext.LogMessages = base.LogEnabledExcludingLogStar("BLIP+")
+	blipContext.LogFrames = base.LogEnabledExcludingLogStar("BLIP++")
+
+	// Create a BLIP-sync context and register handlers:
 	ctx := blipSyncContext{
-		blipContext:       blip.NewContext(),
+		blipContext:       blipContext,
 		dbc:               h.db.DatabaseContext,
 		user:              h.user,
 		effectiveUsername: h.currentEffectiveUserName(),
 	}
-	ctx.blipContext.DefaultHandler = ctx.notFound
+	blipContext.DefaultHandler = ctx.notFound
 	for profile, handlerFn := range kHandlersByProfile {
 		ctx.register(profile, handlerFn)
 	}
 	if !h.db.AllowConflicts() {
 		ctx.register("proposeChanges", (*blipHandler).handleProposedChanges)
 	}
 
-	ctx.blipContext.Logger = func(fmt string, params ...interface{}) {
-		base.LogTo("BLIP", fmt, params...)
-	}
-	ctx.blipContext.LogMessages = base.LogEnabledExcludingLogStar("BLIP+")
-	ctx.blipContext.LogFrames = base.LogEnabledExcludingLogStar("BLIP++")
-
-	// Start a WebSocket client and connect it to the BLIP handler:
-	wsHandler := func(conn *websocket.Conn) {
+	// Create a BLIP WebSocket handler and have it handle the request:
+	server := blipContext.WebSocketServer()
+	defaultHandler := server.Handler
+	server.Handler = func(conn *websocket.Conn) {
 		h.logStatus(101, "Upgraded to BLIP+WebSocket protocol")
 		defer func() {
 			conn.Close()
```

Comment:
If `conn.Close()` returns an error, it's being ignored.  It should log a warning here.